src/ERC4973.sol

```diff
@@ -77,4 +79,18 @@ abstract contract ERC4973 is ERC165, IERC721Metadata, IERC4973 {
 
     emit Revoke(owner, tokenId);
   }
+
+  function _mintWithPermission(
```

Comment:
Ideally, this function is in `ERC4973Permit.sol` and then if the user integrates it, it is `mintWithPermission(...) external`.

Reply:
I actually started out with what you suggested and had to change it because `_mint(...)` is in the context of `ERC4973.sol`. Wdyt?